qf_lib/data_providers/preset_data_provider.py

```diff
@@ -145,51 +145,28 @@ def historical_price(self, tickers: Union[Ticker, Sequence[Ticker]],
                          nr_of_bars: int, end_date: Optional[datetime] = None,
                          frequency: Frequency = None) -> Union[PricesSeries, PricesDataFrame, QFDataArray]:
 
-        assert frequency >= Frequency.DAILY, "Frequency lower than daily is not supported by the Data Provider"
         assert nr_of_bars > 0, "Numbers of data samples should be a positive integer"
         end_date = datetime.now() if end_date is None else end_date
 
         tickers, specific_tickers, tickers_mapping, got_single_ticker = self._tickers_mapping(tickers)
         fields, got_single_field = convert_to_list(fields, PriceField)
         got_single_date = nr_of_bars == 1
 
-        data_bundle = self._data_bundle
-
-        # Data aggregation (allowed only for the Intraday Data and in case if more then 1 data point is found)
-        if frequency < self._frequency and len(data_bundle[DATES]) > 0:
-            data_bundle = self._aggregate_intraday_data_helper(data_bundle, frequency, fields)
-
-        if frequency == Frequency.DAILY:
-            start_date = self._compute_start_date(nr_of_bars, end_date, frequency)
-        else:
-            start_date = end_date - RelativeDelta(days=14)  # to optimize running time for intraday data
+        start_date = self._compute_start_date(nr_of_bars, end_date, frequency)
+        data_bundle = self._data_bundle.loc[start_date:end_date, specific_tickers, fields].dropna(DATES, how='all')
 
-        data_array = data_bundle.loc[start_date:end_date, specific_tickers, fields].dropna(DATES, how='all')
-        data_array = data_array.isel(dates=slice(-nr_of_bars, None))
+        if frequency < self._frequency and data_bundle.shape[0] > 0:  # Aggregate bars to desired frequency
```

Comment:
Do we want to allow aggregating minute bars to daily (or lower) frequency bars?

Reply:
in practice that will rarely be the case, but if someone wants, why not. It will work correctly